server.py

```diff
@@ -116,21 +119,18 @@ def validate():
         if schema is None:
             return client_error("Unsupported schema version '%s'" % version)
 
-        tx_id = None
-        if 'tx_id' in json_data:
-            tx_id = json_data['tx_id']
+        bound_logger = logger.bind(survey_id=json_data.get('survey_id'), tx_id=json_data.get('tx_id'))
 
-        bound_logger = logger.bind(tx_id=tx_id)
         bound_logger.debug("Validating json against schema")
         schema(json_data)
 
         survey_id = json_data.get('survey_id')
-        if survey_id not in KNOWN_SURVEYS[version]:
+        if survey_id not in KNOWN_SURVEYS.get(version, {}):
             bound_logger.debug("Survey id is not known", survey_id=survey_id)
             return client_error("Unsupported survey '%s'" % survey_id)
 
-        instrument_id = json_data['collection']['instrument_id']
-        if instrument_id not in KNOWN_SURVEYS[version][survey_id]:
+        instrument_id = json_data.get('collection').get('instrument_id')
```

Comment:
this will raise a KeyError exception if 'collection' is not a valid key. You should probably default to returning a `{}` instead of `None`.